clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsAccountMapper.java

```diff
@@ -16,71 +16,64 @@
+  final CompositeCredentialsRepository<AccountCredentials> compositeCredentialsRepository;
+  protected final Map<String, String> ecsCredentialsMap;
+  protected final Map<String, String> awsCredentialsMap;
 
   @Autowired
-  public EcsAccountMapper(AccountCredentialsProvider accountCredentialsProvider) {
-    this.accountCredentialsProvider = accountCredentialsProvider;
+  public EcsAccountMapper(
+      @Lazy CredentialsRepository<NetflixECSCredentials> credentialsRepository,
+      @Lazy CompositeCredentialsRepository<AccountCredentials> compositeCredentialsRepository) {
+    this.credentialsRepository = credentialsRepository;
+    this.compositeCredentialsRepository = compositeCredentialsRepository;
 
-    Set<? extends AccountCredentials> allAccounts = accountCredentialsProvider.getAll();
-
-    Collection<NetflixAssumeRoleEcsCredentials> ecsAccounts =
-        (Collection<NetflixAssumeRoleEcsCredentials>)
-            allAccounts.stream()
-                .filter(credentials -> credentials instanceof NetflixAssumeRoleEcsCredentials)
-                .collect(Collectors.toSet());
-
-    ecsCredentialsMap = new HashMap<>();
-    awsCredentialsMap = new HashMap<>();
+    ecsCredentialsMap = new ConcurrentHashMap<>();
+    awsCredentialsMap = new ConcurrentHashMap<>();
+  }
 
-    for (NetflixAssumeRoleEcsCredentials ecsAccount : ecsAccounts) {
-      ecsCredentialsMap.put(ecsAccount.getAwsAccount(), ecsAccount);
+  public void addMapEntry(NetflixAssumeRoleEcsCredentials credentials) {
+    ecsCredentialsMap.put(credentials.getAwsAccount(), credentials.getName());
+    awsCredentialsMap.put(credentials.getName(), credentials.getAwsAccount());
+  }
 
-      allAccounts.stream()
-          .filter(credentials -> credentials.getName().equals(ecsAccount.getAwsAccount()))
-          .findFirst()
-          .ifPresent(
-              v -> awsCredentialsMap.put(ecsAccount.getName(), (NetflixAmazonCredentials) v));
-    }
+  public void removeMapEntry(String ecsAccountName) {
+    ecsCredentialsMap.remove(awsCredentialsMap.get(ecsAccountName));
+    awsCredentialsMap.remove(ecsAccountName);
   }
 
   public NetflixECSCredentials fromAwsAccountNameToEcs(String awsAccountName) {
-    return ecsCredentialsMap.get(awsAccountName);
+    return credentialsRepository.getOne(ecsCredentialsMap.get(awsAccountName));
   }
 
   public NetflixAmazonCredentials fromEcsAccountNameToAws(String ecsAccountName) {
-    return awsCredentialsMap.get(ecsAccountName);
+    return (NetflixAmazonCredentials)
+        compositeCredentialsRepository.getCredentials(
+            awsCredentialsMap.get(ecsAccountName), AmazonCloudProvider.ID);
   }
 
   public String fromAwsAccountNameToEcsAccountName(String awsAccountName) {
-    return ecsCredentialsMap.containsKey(awsAccountName)
-        ? ecsCredentialsMap.get(awsAccountName).getName()
-        : null;
+    return ecsCredentialsMap.get(awsAccountName);
```

Comment:
Previously it would have returned null if `containsKey()` returned false. `get()` returns null if it contains no mapping for the key so not sure if check is still needed here.

Reply:
You're right, and I see the value type has changed as well (to a String from and Obj), so we're no longer concerned about getting an NPE from `getName()`. Cool with leaving as is.